internal/graph/check_fast_path.go

```diff
@@ -443,7 +443,9 @@ func (c *LocalChecker) resolveFastPath(ctx context.Context, leftChans []chan *it
 		if !leftOpen {
 			return
 		}
+		c.goroutineWaiter.Add(1)
 		go func() {
+			defer c.goroutineWaiter.Done()
 			for msg := range leftChan {
```

Comment:
The unit test I wrote asserts that this channel is closed. Otherwise, the `defer` will never run.